docs/reST/tut/newbieguide.rst

```diff
@@ -386,11 +356,14 @@ good that if your code is still slow, and you've done the things I've mentioned
 above, then the problem lies in the way you're addressing your data in python.
 Certain idioms are just going to be slow in python no matter what you do.
 Luckily, python is a very clear language -- if a piece of code looks awkward or
-unwieldy, chances are its speed can be improved, too.  Read over `Python
-Performance Tips`_ for some great advice on how you can improve the speed of
-your code.  That said, premature optimisation is the root of all evil; if it's
-just not fast enough, don't torture the code trying to make it faster.  Some
-things are just not meant to be :)
+unwieldy, chances are its speed can be improved, too.  Read over `Why Pygame is 
+Slow`_ for some deeper insight into why pygame might be considered slower than 
+other frameworks/engines, and what that actually means in practice.  
+And if you're truly stumped by performance problems, profilers like cProfile_ or 
+SnakeViz_ can help identify bottlenecks (they'll tell you which parts of the 
```

Comment:
Technically Snakeviz is a visualizer for the profiles generated by cProfile.

Reply:
"with snakeviz" then.